### PR TITLE
croc: new package

### PR DESCRIPTION
**Pull request by jonasjelonek**
croc is a tool written in Go for sending files from one device to another over the internet using a relay. It runs on multiple platforms, provides end-to-end encryption and works without port forwarding and fixed IP/DynDNS.

Maintainer: me (@jonasjelonek)
Build system: x86_64 Linux
Compile tested: aarch64_cortex-a53, BananaPi R64
Run tested: aarch64_cortex-a53, BananaPi R64

Description:
The description above already summarizes quite well whats the purpose of this tool. A more detailed description can be found [here](https://schollz.com/tinker/croc6/), and there are several examples of its usage in the git [here](https://github.com/schollz/croc).
I use it quite a lot, e.g. when I build new images on my build server (which isn't in my LAN) to transfer it directly to the device to-be-sysupgraded.

### CODE REVIEW

**Comment by PolynomialDivision** [comment]
croc is a very nice tool. Should definitely be added. :)

File: net/croc/Makefile [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=croc
+PKG_VERSION:=9.6.5
+PKG_RELEASE:=1
+
+PKG_SOURCE_PROTO:=git
+PKG_SOURCE_URL:=https://github.com/schollz/croc.git
+PKG_MIRROR_HASH:=324ab1ef98b584904391e55b4f37a1a24fb089303c49a27eeeb4881e7d2d2e82
+PKG_SOURCE_VERSION:=v$(PKG_VERSION)
```
Comment (1715173329) [inline net/croc/Makefile:14]:
```suggestion
PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
PKG_SOURCE_URL:=https://codeload.github.com/schollz/croc/tar.gz/v$(PKG_VERSION)?
PKG_HASH:=2d3ba7bae3c49e3870e2f8523c6be00e92fe6e46828269a8cea34d4034102cad
```

it's recommended to use codeload instead.

File: net/croc/Makefile [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+PKG_LICENSE:=MIT
```
Comment (1715173329) [inline net/croc/Makefile:16]:
```suggestion
PKG_LICENSE:=MIT
PKG_LICENSE_FILES:=LICENSE
```

specify license file

File: net/croc/Makefile [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+PKG_MAINTAINER:=Jonas Jelonek <jelonek.jonas@gmail.com>
+
+PKG_BUILD_DIR:=$(BUILD_DIR)/croc-$(PKG_VERSION)
```
Comment (1715173329) [inline net/croc/Makefile:19]:
```suggestion
```

this is default build path

File: net/croc/Makefile [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+PKG_BUILD_DEPENDS:=golang/host
+PKG_BUILD_PARALLEL:=1
+PKG_BUILD_FLAGS:=no-mips16
+
+GO_PKG:=github.com/schollz/croc
```
Comment (1715173329) [inline net/croc/Makefile:24]:
```suggestion
GO_PKG:=github.com/schollz/croc/v9
GO_PKG_BUILD_PKG:=$(GO_PKG)
```

`go.mod` advertises it's `github.com/schollz/croc/v9`
set `GO_PKG_BUILD_PKG` to build main package (croc) only

File: net/croc/Makefile [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+  transfer files and folders by using a relay.
+endef
+
+define Package/croc/install
+	$(call GoPackage/Package/Install/Bin,$(PKG_INSTALL_DIR))
+
+	$(INSTALL_DIR) $(1)/usr/bin/
+	$(CP) $(PKG_INSTALL_DIR)/usr/bin/croc $(1)/usr/bin/
+endef
+
```
Comment (1715173329) [inline net/croc/Makefile:49]:
```suggestion
```

with `GO_PKG_BUILD_PKG` now we can remove it

**Review by 1715173329 — CHANGES_REQUESTED** [review]

File: net/croc/Makefile (hunk 1 above)
Comment (jonasjelonek) [reply net/croc/Makefile]: done

File: net/croc/Makefile (hunk 3 above)
Comment (jonasjelonek) [reply net/croc/Makefile]: done

File: net/croc/Makefile (hunk 2 above)
Comment (jonasjelonek) [reply net/croc/Makefile:15]: done

File: net/croc/Makefile (hunk 4 above)
Comment (jonasjelonek) [reply net/croc/Makefile]: done

File: net/croc/Makefile (hunk 5 above)
Comment (jonasjelonek) [reply net/croc/Makefile]: done

File: net/croc/Makefile [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+PKG_LICENSE:=MIT
+PKG_LICENSE_FILES:=LICENSE
+PKG_MAINTAINER:=Jonas Jelonek <jelonek.jonas@gmail.com>
+
+PKG_BUILD_DEPENDS:=golang/host
+PKG_BUILD_PARALLEL:=1
+PKG_BUILD_FLAGS:=no-mips16
+
+GO_PKG:=github.com/schollz/croc/v9
+GO_PKG_BUILD_PKG:=$(GO_PKG)
```
Comment (1715173329) [inline net/croc/Makefile:24]:
```suggestion
GO_PKG_BUILD_PKG:=$(GO_PKG)
GO_PKG_LDFLAGS_X:=$(GO_PKG)/src/cli.Version=v$(PKG_VERSION)
```

\+ set build version

Comment (jonasjelonek) [reply net/croc/Makefile:24]: done

**Review by jefferyto — APPROVED** [review]
Thanks!

**Pull request merged**